core/ordersync/ordersync.go

```diff
@@ -111,9 +111,10 @@ type rawResponse struct {
 // Service is the main entrypoint for running the ordersync protocol. It handles
 // responding to and sending ordersync requests.
 type Service struct {
-	ctx          context.Context
-	node         *p2p.Node
-	subprotocols map[string]Subprotocol
+	ctx            context.Context
+	node           *p2p.Node
+	subprotocols   []string
```

Comment:
We also need to use this instead of `SupportedSubprotocols()` when constructing the first request.

```
	return &rawRequest{
		Type:         TypeRequest,
		Subprotocols: s.SupportedSubprotocols(),
		Metadata:     encodedMetadata,
	}, nil
```

Should be:

```
	return &rawRequest{
		Type:         TypeRequest,
		Subprotocols: s.preferredSubprotocols,
		Metadata:     encodedMetadata,
	}, nil
```

Reply:
I updated `SupportedSubprotocols` so that it would be identical to this. Having said this, there's no reason to do the for loop